test/c/test_basis_translator.c

```diff
@@ -0,0 +1,165 @@
+// This code is part of Qiskit.
+//
+// (C) Copyright IBM 2025.
+//
+// This code is licensed under the Apache License, Version 2.0. You may
+// obtain a copy of this license in the LICENSE.txt file in the root directory
+// of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+//
+// Any modifications or derivative works of this code must retain this
+// copyright notice, and modified files need to carry a notice indicating
+// that they have been altered from the originals.
+
+#include "common.h"
+#include <complex.h>
+#include <qiskit.h>
+#include <stdbool.h>
+#include <stddef.h>
+#include <stdint.h>
+#include <stdio.h>
+#include <string.h>
+
+int test_circuit_in_basis(void) {
+    // Create circuit
+    int result = Ok;
+    QkCircuit *circuit = qk_circuit_new(2, 0);
+    qk_circuit_gate(circuit, QkGate_H, (uint32_t[1]){0}, NULL);
+    qk_circuit_gate(circuit, QkGate_CX, (uint32_t[2]){0, 1}, NULL);
+
+    // Create Target compatible with only U gates, with global props.
```

Comment:
```suggestion
    // Create Target already compatible with the circuit.
```

Reply:
Fixed in 5edb0dda618d935b3133572dde9a1266feab2bb7